In [2]:
import pandas as pd
import requests
import json

In [4]:
df = pd.DataFrame()
def get_bikes_in_cities(df):
    for st in df:
        tempdf = pd.DataFrame(st,index = [0])
        tempdf = tempdf[['id','name','latitude','longitude','free_bikes','empty_slots']]
        df = pd.concat([df,tempdf])
    
    df.head()



In [6]:
def GetTips(place):
    url = "https://api.foursquare.com/v3/places/search"
    headers = {"accept": "application/json"}
    response = requests.get(url, headers=headers)
    print(response.text)

    location = response.json[results]
    

In [8]:
# list cities you want to look at here from cities listed at the bottom
cities = ['Nederland']

response = requests.get("http://api.citybik.es/v2/networks")
print(response)
response.json()



<Response [200]>


{'networks': [{'id': 'abu-dhabi-careem-bike',
   'name': 'Abu Dhabi Careem BIKE',
   'location': {'latitude': 24.4866,
    'longitude': 54.3728,
    'city': 'Abu Dhabi',
    'country': 'AE'},
   'href': '/v2/networks/abu-dhabi-careem-bike',
   'company': ['Careem'],
   'gbfs_href': 'https://dubai.publicbikesystem.net/customer/gbfs/v2/en/gbfs.json'},
  {'id': 'acces-velo-saguenay',
   'name': 'Accès Vélo',
   'location': {'latitude': 48.433333,
    'longitude': -71.083333,
    'city': 'Saguenay',
    'country': 'CA'},
   'href': '/v2/networks/acces-velo-saguenay',
   'company': ['PBSC Urban Solutions'],
   'gbfs_href': 'https://saguenay.publicbikesystem.net/customer/gbfs/v2/gbfs.json'},
  {'id': 'aksu',
   'name': 'Aksu',
   'location': {'latitude': 41.1664,
    'longitude': 80.2617,
    'city': '阿克苏市 (Aksu City)',
    'country': 'CN'},
   'href': '/v2/networks/aksu',
   'company': ['阿克苏公共服务']},
  {'id': 'alba',
   'name': 'Alba',
   'location': {'latitude': 44.716667,
    'longitude': 

In [32]:

get_bikes_in_cities(df)

In [34]:
response = requests.get('http://api.citybik.es/v2/networks/ov-fiets')


In [62]:

r = response.json()
q = r['network']
print(q.keys())
print(type(q))
r

dict_keys(['id', 'name', 'location', 'href', 'company', 'source', 'stations'])
<class 'dict'>


{'network': {'id': 'ov-fiets',
  'name': 'OV-fiets',
  'location': {'latitude': 52.1449,
   'longitude': 5.5603,
   'city': 'Nederland',
   'country': 'NL'},
  'href': '/v2/networks/ov-fiets',
  'company': ['NS Groep NV'],
  'source': 'https://apiportal.ns.nl/',
  'stations': [{'id': '009ac5a5dfe1e13da61ea613ebd580dd',
    'name': 'Alkmaar',
    'latitude': 52.63753,
    'longitude': 4.73985,
    'timestamp': '2025-01-30T22:08:29.637411Z',
    'free_bikes': 127,
    'empty_slots': None,
    'extra': {'uid': 'amr001'}},
   {'id': '00deaf43155f1d6c57891732c5434185',
    'name': 'Amsterdam Noord',
    'latitude': 52.402374,
    'longitude': 4.931019,
    'timestamp': '2025-01-30T22:08:29.636594Z',
    'free_bikes': 15,
    'empty_slots': None,
    'extra': {'uid': 'asd009'}},
   {'id': '04672c55a75fa3cdef4533bd1277ec53',
    'name': 'Hoogeveen',
    'latitude': 52.73309,
    'longitude': 6.47358,
    'timestamp': '2025-01-30T22:08:29.638079Z',
    'free_bikes': 39,
    'empty_slots': None

In [98]:
list = []
# for item in r['network']:
#     print(item['id'])
    # df['id'].append(item.get('id'))
#'id','name','latitude','longitude','free_bikes','empty_slots'
for station in r["network"]["stations"]:  # Loop over the list inside "stations"
    list.append(pd.json_normalize(station))

In [119]:
station_ids = [station["id"] for station in r["network"]["stations"]]
station_names = [station["name"] for station in r["network"]["stations"]]
latitudes = [station["latitude"] for station in r["network"]["stations"]]
longitudes = [station["longitude"] for station in r["network"]["stations"]]
free_bikes = [station["free_bikes"] for station in r["network"]["stations"]]
# Create a DataFrame
df = pd.DataFrame({
    "Station ID": station_ids,
    "Name": station_names,
    "Latitude": latitudes,
    "Longitude": longitudes,
    "Free Bikes": free_bikes
})
df

Station ID                       Name   Latitude  \
0    009ac5a5dfe1e13da61ea613ebd580dd                    Alkmaar  52.637530   
1    00deaf43155f1d6c57891732c5434185            Amsterdam Noord  52.402374   
2    04672c55a75fa3cdef4533bd1277ec53                  Hoogeveen  52.733090   
3    0725be1791baf25a0b247bf340560069  Zoetermeer Driemanspolder  52.049090   
4    07b969ad47b4381c47b2eaa0a50d3d30         Veenendaal Centrum  52.019787   
..                                ...                        ...        ...   
280  fd45aafc62edaa3a2759fc5ae6b83dea              Breda Centrum  51.595020   
281  fd61144b1ba7d7af027695e286da43eb           Zwolle Hanzeland  52.504990   
282  fdaf68bb8601c63dd8f165750f5d20dd                     Beilen  52.854626   
283  fdceea2faa8e8b1c5abef210cebdb7d1              Horst Sevenum  51.427600   
284  fe44fdb504132575b0c631f8f3ba4d80              Anna Paulowna  52.867532   

     Longitude  Free Bikes  
0     4.739850         127  
1     4.931019          15  
2     6.473580          39  
3     4.476630          18  
4     5.549334           3  
..         ...         ...  
280   4.779437         159  
281   6.091060          22  
282   6.520572          12  
283   6.040800          11  
284   4.811490           8  

[285 rows x 5 columns]

In [116]:
df.to_csv("df_city_bikes.csv")

1 Abu Dhabi
2 Saguenay
3 阿克苏市 (Aksu City)
4 Alba
5 Albacete
6 Almeirim
7 Almaty
8 León
9 Badalona
10 Castelldefels
11 Cornellà de Llobregat
12 El Prat de Llobregat
13 Esplugues de Llobregat
14 Gavà
15 Molins de Rei
16 Sant Boi de Llobregat
17 Sant Joan Despí
18 Sant Just Desvern
19 Santa Coloma de Gramenet
20 Viladecans
21 Andria
22 阿拉尔市 (Aral)
23 Arezzo
24 Nitra
25 Scandicci
26 Ascoli Piceno
27 Athens
28 Austin, TX
29 Auxerre
30 Québec
31 Aventura, FL
32 Avilés
33 Ahrweiler
34 Antalya
35 Çanakkale
36 Nilüfer
37 Batumi
38 San Francisco Bay Area, CA
39 Aller
40 Belfast
41 Bentonville, AR
42 Bergen
43 Bergisches e-Bike
44 Bournemouth, Christchurch and Poole
45 Brighton
46 Cornwall
47 Greater Manchester
48 Hackney
49 Hereford
50 Hertsmere
51 Isle of Wight
52 Leeds
53 Norwich
54 Plymouth
55 Portsmouth
56 Southampton
57 Watford
58 West Midlands
59 Westminster
60 Wool
61 Ovar
62 Bianzone
63 Badajoz
64 Boadilla del Monte
65 Bolzano
66 Como
67 Ferrol - Narón
68 Busto Arsizio
69 Perugia
70 Alcázar de San Juan
71 A Illa de Arousa
72 Arteixo
73 Burgos
74 Castelló de la Plana
75 Fortaleza
76 A Coruña
77 Cuenca
78 Elx
79 Ljubljana
80 Logroño
81 Madrid
82 Brescia
83 Mislata
84 Acquarica del Capo
85 Acquaviva delle fonti
86 Alghero
87 Assemini
88 Borgarello
89 Brancaleone
90 Carrara
91 Casaletto Spartano
92 Castel Guelfo di Bologna
93 Cesena
94 Cisterna di Latina
95 Montesilvano
96 Elmas
97 Parco Nazionale del Circeo
98 Ente Parco Montemarcello Magra
99 Parco naturale Migliarino San Rossore Massaciuccoli
100 Forlì
101 Gaeta
102 Gallipoli
103 Imola
104 Lavis
105 Livorno
106 Mantova
107 Massa
108 Mezzocorona
109 Mezzolombardo
110 Modena
111 Monserrato
112 Montecatini Terme
113 Monteriggioni
114 Monza
115 Morigerati
116 Parco dei Colli di Bergamo
117 Parco Naturale Molentargius - Saline
118 Taranto
119 Pizzighettone e Formigara
120 Porto Torres
121 Rivarolo Canavese
122 San Donato Milanese
123 San Giovanni in Persiceto
124 San Lazzaro di Savena
125 San Michele All'adige
126 San Teodoro
127 Scafati
128 Scandiano
129 Selargius
130 Siena
131 TV
132 Tortolì
133 Treviglio
134 Trieste
135 Vicolungo
136 Barcelona
137 Quartucciu
138 Rivas-Vaciamadrid
139 Palma
140 Sant Vicent del Raspeig
141 Schio
142 Leiria
143 Nantes
144 Prešov
145 Biella
146 Bergamo
147 Chattanooga, TN
148 Curitiba
149 Nordelta
150 Brasília
151 Žilina
152 Milano
153 Campoo de Enmedio
154 Porto Alegre
155 Recife
156 Rio de Janeiro
157 Salvador
158 São Paulo
159 Santiago
160 Santos
161 Península de Papagayo
162 Portland, OR
163 Beaverton, OR
164 Bydgoszcz
165 Verona
166 Vitória
167 Valladolid
168 Bilbao
169 Bismarck, ND
170 Montréal, QC
171 Toronto, ON
172 Zaragoza
173 Barakaldo
174 Berango
175 Bilbao
176 Erandio
177 getxo
178 Leioa
179 Portugalete
180 Santurtzi
181 Sestao
182 Banja Luka
183 Boston, MA
184 CIM Viseu Dão-Lafões
185 Boulder, CO
186 Brindisi
187 Fort Lauderdale, FL
188 Budapest
189 Milwaukee, WI
190 Aveiro
191 Neusiedler See
192 Lillestrøm
193 Clermont-Ferrand
194 Cagliari
195 Berlin
196 Darmstadt
197 Frankfurt
198 Köln
199 München
200 Washington, DC
201 Carbonia
202 Switzerland
203 Cascavel
204 Cassano d'Adda
205 Castellucchio
206 La Chantrerie, Nantes
207 Charlotte, NC
208 Bhopal
209 Bilaspur
210 Moradabad
211 Prayagraj
212 Rajarhat New Town
213 Ranchi
214 Srinagar
215 Surat
216 Chivasso
217 Cholet
218 Pisa
219 Cincinnati, OH
220 Miami, FL
221 New York, NY
222 Lima
223 Linz
224 Helsinki
225 Civitavecchia
226 Clemson, SC
227 Cluj-Napoca
228 Exeter
229 Columbus, OH
230 Cork
231 Cuneo
232 Ardenne Métropole
233 Carcassonne
234 Aigialeia
235 Αrxaia Olympia
236 Chania
237 Edessa
238 Florina
239 Igoumenitsa
240 Kιato
241 Limnos
242 Loutraki
243 Marathon
244 Maroussi
245 Μoschato-Tavros
246 Nafplio
247 Νea Smyrni
248 Rhodes
249 Syros
250 Soissons
251 Wageningen
252 Donostia/San Sebastián
253 Mannheim
254 Denver, CO
255 Des Moines, IA
256 Dijon
257 Chicago, IL
258 愛知県 (Aichi)
259 福井県 (Fukui)
260 岐阜県 (Gifu)
261 広島県 (Hiroshima)
262 兵庫県 (Hyōgo)
263 茨城県 (Ibaraki)
264 石川県 (Ishikawa)
265 鹿児島県 (Kagoshima)
266 神奈川県 (Kanagawa)
267 京都府 (Kyoto)
268 茨城県 (Miyagi)
269 奈良県 (Nara)
270 新潟県 (Niigata)
271 大分県 (Ōita)
272 岡山県 (Okayama)
273 沖縄県 (Okinawa)
274 大阪府 (Osaka)
275 滋賀県 (Shiga)
276 Tokyo
277 山梨県 (Yamanashi)
278 Gent
279 Dubai
280 Dublin
281 Rovereto
282 Trento
283 Didymoteicho
284 Glyfada
285 Igoumenitsa
286 Ioannina
287 Irakleio
288 Karditsa
289 Kinouria
290 Komotini
291 Mani
292 Patra
293 Rethymno
294 Samos
295 Cd de México
296 Buenos Aires
297 Grünheide (Mark)
298 Eifel e-Bike
299 El Paso, TX